app/utils/transformers/DepartureTransportMeansTransformer.scala

```diff
@@ -19,26 +19,32 @@ package utils.transformers
 import generated.DepartureTransportMeansType02
 import models.{Index, UserAnswers}
 import pages._
-import pages.departureMeansOfTransport.{CountryPage, VehicleIdentificationNumberPage}
+import pages.departureMeansOfTransport.{CountryPage, TransportMeansIdentificationPage, VehicleIdentificationNumberPage}
+import services.MeansOfTransportIdentificationTypesService
+import uk.gov.hmrc.http.HeaderCarrier
 
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
-class DepartureTransportMeansTransformer @Inject() (implicit ec: ExecutionContext) extends PageTransformer {
+class DepartureTransportMeansTransformer @Inject() (meansOfTransportIdentificationTypesService: MeansOfTransportIdentificationTypesService)(implicit
+  ec: ExecutionContext
+) extends PageTransformer {
 
-  def transform(departureTransportMeans: Seq[DepartureTransportMeansType02]): UserAnswers => Future[UserAnswers] =
+  def transform(departureTransportMeans: Seq[DepartureTransportMeansType02])(implicit headerCarrier: HeaderCarrier): UserAnswers => Future[UserAnswers] =
```

Comment:
To ensure that the reference data lookups happen in parallel, could we maybe do something like
```
  def transform(departureTransportMeans: Seq[DepartureTransportMeansType02])(implicit headerCarrier: HeaderCarrier): UserAnswers => Future[UserAnswers] =
    userAnswers => {
      lazy val referenceDataLookups = departureTransportMeans.map {
        dtm =>
          referenceDataConnector.getMeansOfTransportIdentificationType(dtm.typeOfIdentification).map((dtm, _))
      }

      Future.sequence(referenceDataLookups).flatMap {
        _.zipWithIndex.foldLeft(Future.successful(userAnswers))({
          case (acc, ((DepartureTransportMeansType02(_, _, identificationNumber, nationality), typeOfIdentification), i)) =>
            acc.flatMap {
              userAnswers =>
                val dtmIndex = Index(i)
                val pipeline = set(TransportMeansIdentificationPage(dtmIndex), typeOfIdentification) andThen
                  set(VehicleIdentificationNumberPage(dtmIndex), identificationNumber) andThen
                  set(CountryPage(dtmIndex), nationality)

                pipeline(userAnswers)
            }
        })
      }
    }
```